### ISSUE TITLE
Help with WGS data

### ISSUE THREAD

**Issue by mdiaz09** [NatalieKAndersson/DEVOLUTION#2]
Hello,

I am using WGS data after running HATCHet and DeCiFer on paired single biopsy tumors, which outputs a datatable with CCF, VAF, and clone estimates. I am trying to determine a phylogeny but keep running into the following error after DEVOLUTION finds ~ 505 objects after DBSCAN 

`Error in `[<-`(`*tmp*`, t, 1, value = paste("Cluster", t)) : 
  subscript out of bounds`

DEVOLUTION seems to do better when running with a higher event_co but as each mutation is an SNP I am not able to use a high value. 
Would you be able to provide me some tips for how to process/analyze WGS data?  Attached is a few rows of the data file that I am using, with the clones identified by DeCiFer being relabeled as different Samples.

[test_head.xlsx](https://github.com/NatalieKAndersson/DEVOLUTION/files/9660613/test_head.xlsx)

**Comment by NatalieKAndersson** [comment]
Hello!

Thank you for using DEVOLUTION. I'm happy to answer the questions.

**event_co:** It will not help you so much with the clustering. This is mainly used to determine if two genetic alterations with a bit different different start and/or end positions, actually are the same event and differ a bit in their start/end position due to noise. So, if you are certain that all the start- and end positions are entirely correct you should put 1 there. Otherwise increase it a bit to cluster together events a bit more.

![image](https://user-images.githubusercontent.com/71607509/192961159-24411469-4d53-41f7-b3dc-37b0e0c8c229.png)

I think what you want to do is to increase the eps argument. This makes the clustering with dbscan a bit more generous. If eps is set too low, it might consider even very small differences between genetic alterations, as different clusters. You often want it to be at the "elbow" of the plot. So, try to increase it. It is done in the devolution function.

DEVOLUTION(datasegment,event_co,datatypes=c("All"), **eps = 0.5**, names="letters")

Not at the elbow (eps is in the y-axis)
![image](https://user-images.githubusercontent.com/71607509/192971741-1dcb27b5-26da-4099-b6df-01171d5a5314.png)

Closer to the elbow
![image](https://user-images.githubusercontent.com/71607509/192971919-64634cc1-b244-4a0d-b0fb-e30dbcad2a55.png)


Another thing that could be helpful for the clustering would be to round the CCF values, to nearest whole number or 5 or 10 since we do not have that measurement certainty to have that many decimals.
I.e. 35,0729504244428 --> 35

In this column you might also consider defining what the alteration in that position is, such as A>G.

![image](https://user-images.githubusercontent.com/71607509/192966102-feb9e866-1e32-45a8-b769-cd008a9640a3.png)

Let me know if this helped. You can also reach me on natalie.andersson@med.lu.se

Best regards
Natalie

**Comment by mdiaz09** [comment]
I appreciate the response! After increasing the elbow score I have encountered a different error.

```
Error in `[<-`(`*tmp*`, s, 1, value = file_new[k, 11]) : 
  subscript out of bounds
```

**Comment by NatalieKAndersson** [comment]
Hello!

How does your elbow plot look? Maybe 3 is too much in your case, since almost all are in the same cluster?

You also need to have identifiers for the genetic alterations in the "Cytoband/ Gene" column. The program compares the name as well as the position, when considering if an event across samples are "the same" or not. If there is no name for the alterations, it will create a name like SNV v1, and you have thousands of rows so I suspect that it might have run out of the names -> file_new is not created properly

Best regards
Natalie

**Comment by mdiaz09** [comment]
When running with 100 event_co I get the following elbow plot. I adjusted the eps after re-reading your comment, but my data set doesn't seem to have a true elbow. I also edited the "Cytoband/ Gene" column to show the nucleotide alteration (ie A>T) but am still getting the same out of bounds error after running the DEVOLUTION function. 

```Error in `[<-`(`*tmp*`, s, 1, value = file_new[k, 11]) : 
  subscript out of bounds```

<img width="583" alt="Screen Shot 2022-09-29 at 12 49 17 PM" src="https://user-images.githubusercontent.com/95503761/193128139-15368694-1c7a-4be0-9e97-2176a2f76ecf.png">

It also may be worth noting that I have ~50 clusters identified
